zhmccli/_cmd_user_pattern.py

```diff
@@ -0,0 +1,674 @@
+    * via LDAP group lookup (--ldap-group-template-... options)
+
+    For some options, the corresponding HMC properties are different than
+    the option names. In such cases, the option descriptions mention the
+    corresponding HMC property names.
+
+    There are some old properties that are set by the HMC consistent with
+    the preferred new properties. Because of that, the old properties have
+    no corresponding options and the options for the new preferred properties
+    should be used instead. The old properties are:
+
+    \b
+    * ldap-server-definition-uri:
+      Name of LDAP Server Definition object used for LDAP attribute lookup for
+      specific user template name and domain name restrictions. Use the
+      --ldap-template-lsd and --domain-name-restrictions-lsd options instead.
+    * user-template-uri:
+      Name of the specific user template to be used. Use the
+      --specific-template option instead.
+
+    In addition to the command-specific options shown in this help text, the
+    general options (see 'zhmc --help') can also be specified right after the
+    'zhmc' command name.
+    """
+    cmd_ctx.execute_cmd(lambda: cmd_user_pattern_create(cmd_ctx, options))
+
+
+@user_pattern_group.command('update', options_metavar=COMMAND_OPTIONS_METAVAR)
+@click.argument('USER_PATTERN', type=str, metavar='USER_PATTERN')
+@click.option('--name', type=str, required=False,
+              help='The new name of the user pattern.')
+@click.option('--description', type=str, required=False,
+              help='The new description of the user pattern.')
+@click.option('--pattern', type=str, required=False,
+              help='The new pattern for matching userids. '
+              'The combination of "pattern" and "type" must be unique.')
+@click.option('--type', type=click.Choice(['glob-like', 'regular-expression']),
+              required=False,
+              help='The new style in which the pattern is expressed.')
+@click.option('--retention-time', type=int, required=False, default=1,
+              help='The new time in days that the pattern-based users created '
+              'based on this user pattern will be retained after being '
+              'created. '
+              'A value of 0 indicates not to retain the users after being '
+              'created.')
+@click.option('--specific-template', type=str, required=False,
+              # HMC property: 'specific-template-uri/name'
+              help='Direct template identification: '
+              'The new name of the user template to be used. '
+              'The empty string disables this template identification method. '
+              'HMC property: specific-template-uri/name')
+@click.option('--ldap-template-lsd', type=str, required=False,
+              # HMC property: 'template-name-override-ldap-server-definition-
+              #   uri/name'
+              help='LDAP attribute lookup based template identification: '
```

Comment:
I am wondering if you could at least put the repeated help texts into a single source to avoid duplication of that content.

Reply:
They are not repeated. There are differences between create and update for each option. For example, create defines a default but update does not (because the default is not to change). Update defines in cases where the property can be null how to reset it (usually with empty string), while in create that is not important.